static/js/import_wizard/utils/obs_generation.ts

```diff
@@ -54,24 +55,35 @@ function hasRequiredProps(obs: Observation): boolean {
 function generateObservationsInRow(
   row: Array<string>,
   header: Array<Column>,
-  obsGenMaps: ObsGenMaps
+  obsGenMaps: ObsGenMaps,
+  valueMap: ValueMap
 ): Array<Observation> {
   const obsList = new Array<Observation>();
   for (const valColIdx of Array.from(obsGenMaps.valCol2Hdr.keys())) {
     const hdrThing = obsGenMaps.valCol2Hdr.get(valColIdx);
     const obs: Observation = new Map();
+    let cellVal = row[valColIdx] || "";
+    const lowerCaseCellVal = cellVal.toLowerCase();
```

Comment:
Given this happens twice, use a helper `getCellVal(row, valColIdx, valueMap)` that does the empty check, lower-casing and valueMap lookup if needed?

Reply:
done